### PR TITLE
Update QuestionAnswering.php

### PR DESCRIPTION
**Pull request by cirolaferrara**
When VectorStoreBase is void, $systemMessageTemplate is ignored

### CODE REVIEW

**Comment by f-lombardo** [comment]
Two small annotations:
1. running `composer refactor` cold solve the problems in CI
2. is it possible to write a test for the issue you are addressing?

Thank you.

**Comment by cirolaferrara** [comment]
My first test :) check it

**Comment by f-lombardo** [comment]
Hi @cirolaferrara , I think that your PR solves a real probklem, thanks! 
I created a fork that fixes some minor issues in your PR: https://github.com/f-lombardo/LLPhant/tree/ciro
I think that you can evaluate my changes and , if you agree, you can copy them on your PR. What do you think?

**Comment by cirolaferrara** [comment]
Done. Thanks for the improvements

File: tests/OpenAIChatWithSystemMessageTemplateTest.php
Comment (f-lombardo) [inline tests/OpenAIChatWithSystemMessageTemplateTest.php:1]: There is a test more on less on the same topic in `tests/Integration/Query/SemanticSearch`. Maybe also this one could be moved into that directory

**Comment by MaximeThoonsen** [comment]
Thanks @cirolaferrara and @f-lombardo !

**Pull request merged**